src/sdk/config_fs.js

```diff
@@ -151,88 +224,200 @@ class ConfigFS {
+     * {config_dir}/identities/{account_id}/
+     * @param {string} id
+     * @returns {string} 
+    */
+    get_identity_dir_path_by_id(id) {
+        return path.join(this.config_root, CONFIG_SUBDIRS.IDENTITIES, id, '/');
+    }
+
+    /**
+     * get_account_or_user_path_by_access_key returns the full account path by access key as follows
+     * {config_dir}/access_keys/{access_key}.symlink
      * @param {string} access_key
      * @returns {string} 
      */
-    get_account_path_by_access_key(access_key) {
+    get_account_or_user_path_by_access_key(access_key) {
         return path.join(this.access_keys_dir_path, this.symlink(access_key));
     }
 
     /**
-     * get_old_account_path_by_name returns the full account path by name based on old config dir structure
+     * _get_old_account_path_by_name returns the full account path by name based on old config dir structure
+     * as follows - {config_dir}/accounts/{access_name}.json
      * @param {string} account_name
      * @returns {string} 
     */
-    get_old_account_path_by_name(account_name) {
-        return path.join(this.accounts_dir_path, this.json(account_name));
+    _get_old_account_path_by_name(account_name) {
+        return path.join(this.old_accounts_dir_path, this.json(account_name));
     }
 
     /**
-     * is_account_exists returns true if account config path exists in config dir
-     * it can be determined by any account identifier - name, access_key and in the future id
-     * @param {{ name?: string, access_key?: string }} identifier_object
+     * is_account_exists_by_name returns true if account config path exists in config dir
+     * if account does not exist and it's a regular account (not an IAM user) 
+     * try to locate it under the old accounts/ directory
+     * @param {string} account_name
+     * @param {string} [owner_account_id]
      * @returns {Promise<boolean>} 
     */
-    async is_account_exists(identifier_object) {
-        if (!identifier_object || typeof identifier_object !== 'object') throw new Error('config_fs.is_account_exists - no identifier provided');
-       let path_to_check;
-       let use_lstat = false;
-       if (identifier_object.name) {
-           // TODO - when users/ move to be symlink we need to use_lstat by name, id is not using lstat
-           path_to_check = this.get_account_path_by_name(identifier_object.name);
-        } else if (identifier_object.access_key) {
-            path_to_check = this.get_account_path_by_access_key(identifier_object.access_key);
-            use_lstat = true;
-       } else {
-            throw new Error(`config_fs.is_account_exists - invalid identifier provided ${identifier_object}`);
+    async is_account_exists_by_name(account_name, owner_account_id) {
+        const path_to_check = this.get_account_or_user_path_by_name(account_name, owner_account_id);
+        let res = await native_fs_utils.is_path_exists(this.fs_context, path_to_check);
```

Comment:
For easier readability of the next condition, I would change the name of the variable "res" to a more meaningful name.